File: SpringBootLignesTelephoniques/src/main/java/com/telephone/backendlignestelephoniques/entities/LigneTelephonique.java
package com.telephone.backendlignestelephoniques.entities;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.telephone.backendlignestelephoniques.enums.EtatType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.EqualsAndHashCode;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;
import org.springframework.data.annotation.CreatedDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class LigneTelephonique {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long idLigne;

    @Column(unique = true, nullable = false)
    private String numeroLigne;

    private String affectation;
    private String poste;

    @Enumerated(EnumType.STRING)
    private EtatType etat;

    private Date dateLivraison;

    //@Column(unique = true, nullable = false)
    private String numeroSerie;

    private Double montant;

    @CreatedDate
    private Date createdDate;

    @Column(name = "id_type_ligne", insertable = false, updatable = false)
    private Long typeId;

    @ManyToOne
    @JoinColumn(name = "id_type_ligne")
    @OnDelete(action = OnDeleteAction.CASCADE)
    private TypeLigne typeLigne;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "ligne_id", referencedColumnName = "idLigne")
    private Set<LigneAttribut> ligneAttributs = new HashSet<>();

}

File: AngularTelephoneLines/src/app/Models/LigneTelephonique.ts
import {TypeLigne} from "./TypeLigne";
import {LigneAttribut} from "./LigneAttributs";

export interface LigneTelephonique {
  idLigne?: number;
  typeId?: number;
  numeroLigne: string;
  affectation?: string;
  poste?: string;
  etat?: EtatType | null;
  dateLivraison?: Date | null;
  numeroSerie?: string;
  montant?: number;
  createdDate: Date;
  typeLigne: TypeLigne;
  ligneAttributs?: any[];
}

export enum EtatType {
  NULL = "NULL",
  INACTIF= "INACTIF",
  ACTIF = "ACTIF",
  RESILIE = "RESILIE",
  CESSION = "CESSION"
}
